src/public-lib/components/checkedSelect/CheckedSelect.tsx

```diff
@@ -86,6 +90,19 @@ export default class CheckedSelect extends React.Component<CheckedSelectProps, {
     {
         return getOptionLabel(option, this.selectedValues, this.props.checkBoxType);
     }
+    
+    @computed get inputValue() {
+        return this.props.inputValue || this.defaultInputValue;
+    }
+    
+    @autobind
+    @action defaultOnInputChange(input:string, options:{action:string}) {
+        // to prevent clearing input value when selecting an option
+        if (options.action !== "set-value") {
```

Comment:
can you explain more whats going on/why this is necessary?

Reply:
The change events of input in react-select include: `set-value, input-change, input-blur, menu-close`. On the `set-value` event, an option is selected from the list and a blank string will be updated to the search input by default. We want to prevent clearing the search input value after selecting an option in order to keep the filtered list open for the users to select more than one items from the list.

Reply:
what confuses me is where the blank string stuff is done here. in the call to this.props.onInputChange? if so then this.props.onInputChange should have a different name that makes clear that it always will clear the value. otherwise, you are tailoring a general library function to a very specific use

Reply:
Without the `if` condition, the input value (which will be a blank string in the case of `action === 'set-value'`) will always be passed to `this.props.onInputChange`. This leads to undesirable behaviour so adding the `if` condition will prevent that.

Reply:
oh I see. that makes sense. can you put that in the comment?